File: 프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Deque<int[]> q = new ArrayDeque<>();
        
        
        // {우선순위, 인덱스}
        for(int i=0; i<priorities.length; i++)
            q.addLast(new int[]{priorities[i], i});
        
        
        
        while(q.size() > 0){
            // 우선순위가 높은 프로세스 확인
            int[] mx = q.stream()
                .max(Comparator.comparingInt(v -> v[0])).orElse(null);
            
            
            int stopped_idx = -1;
            while(true){
                int[] p = q.removeFirst();
                
                if(p[0] == mx[0]){
                    answer++;
                    stopped_idx = p[1];
                    break;
                }
                
                q.addLast(p);
            }
            
            if(stopped_idx == location)
                return answer;
        }
        
        
        return -1;
    }
}

File: 프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    
    
    # (값, 인덱스) 순으로 큐에 push
    
    q = deque()
    for idx, order in enumerate(priorities):
        q.append((order, idx))


    while q:
        # q 배열에서 우선순위(1 ~ 9)가 가장 높은 작업 찾기
        mx, i = max(q)
        
        stopped_idx = -1
        # 우선순위가 높은 작업 찾아 실행
        while True:
            p, idx = q.popleft()
            
            # 우선순위가 낮은 경우
            if p != mx:
                q.append((p, idx))
            
            # 우선순위가 높아 실행할 프로세스일 경우
            if p == mx:
                answer += 1
                stopped_idx = idx
                break
                
                
        # 실행한 프로세스가 찾는 프로세스일 경우
        if stopped_idx == location:
            return answer
                
